### ISSUE TITLE
macroexecutor duplicate button does not takes max repetitions into account

### ISSUE THREAD

**Issue by mrosanes** [sardana-org/sardana#1104]
For a param repeat node, macroexecutor duplicate button does not take the number of maximum repetitions into account. The node can be repeated more times than the maximum number or repeats allowed. This should be limited.

Moreover, a popup does not appear if the max repetitions is exceeded in macroexecutor (spock yes, it reports that the max has been surpassed). The macro seems to be launched correctly from macroexecutor, but it is not, because the max repetitions is surpassed.

It can be tested easily with 'pt8' from sardana example parameters macro.

**Comment by teresanunez** [comment]
I have just tried to reproduce this (using pt8 macro, where the maximum number of repetitions is set to 2) and the macroexecutor does not allow me to add more than two repetitions. So in my case it works like it should (well, I do not get any pop-up window telling why not more repetitions are added ... but they are not added).

**Comment by mrosanes** [comment]
Hi @teresanunez,
Have you tried adding more repetitions with the duplicate button? (not the '+' button, but the 'copy' button)

**Comment by teresanunez** [comment]
Sorry, I was not aware of the copy button, i tried it with the Add one.

**Comment by mrosanes** [comment]
> Sorry, I was not aware of the copy button, i tried it with the Add one.

Yes, this copy button is useful above all, in case of macros with a lot of parameters on a repeat node, and having to use the same values for a high number of these parameters on the different repetitions of the node.

The node can be duplicated many times in macroexecutor using the copy button, and only change the few parameters needed to be changed. 

In ALBA we are using it a lot in a Tomography Microscopy Beamline that needs to generate scripts that controls a X-Ray Microscope: this microscope only accepts txt scripts for giving the commands (move lens, move sample, collect, etc); the copy button for repeat parameters is really useful to generate txt scripts for collecting different regions of a sample. It allows to diminish the number of mistakes when users enters the parameter values when generating the txt script.
